platforms/core-runtime/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/codegen/jvmbytecode/InterceptJvmCallsGenerator.java

```diff
@@ -132,20 +156,86 @@ private static TypeSpec generateFactoryClass(String className, BytecodeIntercept
 
 
     private static void generateVisitMethodInsnCode(
-        MethodSpec.Builder visitMethodInsnBuilder,
-        Collection<CallInterceptionRequest> interceptionRequests,
+        MethodSpec.Builder method,
         Map<Type, FieldSpec> typeFieldByOwner,
+        Map<CallableOwnerInfo, List<CallInterceptionRequest>> requestsByOwner,
         Consumer<? super CallInterceptionRequest> onProcessedRequest,
         Consumer<? super FailureInfo> onFailure
     ) {
         CodeBlock.Builder code = CodeBlock.builder();
-        interceptionRequests.stream()
-            .collect(Collectors.groupingBy(it -> it.getInterceptedCallable().getOwner(), LinkedHashMap::new, Collectors.toList()))
-            .forEach((owner, requests) -> generateCodeForOwner(owner, typeFieldByOwner, requests, code, onProcessedRequest, onFailure));
+        requestsByOwner.forEach((owner, requests) -> generateCodeForOwner(
+            owner,
+            typeFieldByOwner,
+            requests,
+            code,
+            InterceptJvmCallsGenerator::matchOpcodeExpression,
+            InterceptJvmCallsGenerator::generateInterceptedInvocation,
+            InterceptJvmCallsGenerator::generateKotlinDefaultInvocation,
+            onProcessedRequest,
+            onFailure));
         code.addStatement("return false");
-        visitMethodInsnBuilder.addCode(code.build());
+        method.addCode(code.build());
+    }
+
+    private static void generateFindBridgeMethodBuilderCode(
```

Comment:
❓ Do you think it makes sense  we also add annotation processor tests for this, to see what gets generated?

Reply:
>Do you think it makes sense we also add annotation processor tests for this, to see what gets generated?

Done.